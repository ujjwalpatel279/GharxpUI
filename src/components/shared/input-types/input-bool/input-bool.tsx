import React, { forwardRef, useEffect, ChangeEvent } from 'react';
import { useForm } from 'react-hook-form';
import { InputListItem } from '../input-list-item/input-list-item';

interface InputBoolProps {
  name: string;
  legend: string;
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  disabled?: boolean;
}

type Ref = HTMLInputElement;

export const InputBool = forwardRef<Ref, InputBoolProps>(
  ({ name, legend, setValue, value, onChange, disabled }, ref) => {
    useEffect(() => {
      if (value) {
        setValue(name, value);
      }
    }, [value]);

    return (
      <fieldset>
        <legend>{legend}</legend>
        <div className="l-flex l-flex--wrap">
          <div>
            <InputListItem
              type="radio"
              id={`yes-${name}`}
              name={name}
              value="Yes"
              ref={ref}
              onChange={onChange}
              disabled={disabled}
            />
          </div>
          <div>
            <InputListItem
              type="radio"
              id={`no-${name}`}
              name={name}
              value="No"
              ref={ref}
              onChange={onChange}
              disabled={disabled}
            />
          </div>
        </div>
      </fieldset>
    );
  },
);

InputBool.displayName = 'InputBool';
