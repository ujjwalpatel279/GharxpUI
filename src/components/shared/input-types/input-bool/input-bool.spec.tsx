import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputBool } from './input-bool';

describe('Input Bool Component', () => {
  afterEach(cleanup);

  it('renders the bool component', () => {
    const { container } = render(<InputBool name="inputbool" legend="Bool legend" ref={() => null} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
