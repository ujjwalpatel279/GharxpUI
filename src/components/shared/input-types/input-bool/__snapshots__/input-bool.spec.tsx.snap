// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Bool Component renders the bool component 1`] = `
<fieldset>
  <legend>
    Bool legend
  </legend>
  <div
    class="l-flex l-flex--wrap"
  >
    <div>
      <label
        class="c-input-list-item"
        for="yes-inputbool"
      >
        <input
          id="yes-inputbool"
          name="inputbool"
          type="radio"
          value="Yes"
        />
        <span
          class="c-radio-input__inner"
        >
          <span>
            Yes
          </span>
        </span>
      </label>
    </div>
    <div>
      <label
        class="c-input-list-item"
        for="no-inputbool"
      >
        <input
          id="no-inputbool"
          name="inputbool"
          type="radio"
          value="No"
        />
        <span
          class="c-radio-input__inner"
        >
          <span>
            No
          </span>
        </span>
      </label>
    </div>
  </div>
</fieldset>
`;
